### PR TITLE
UI navigation system

### PR DESCRIPTION
**Pull request by FerchusGames**
https://github.com/user-attachments/assets/26764833-d7a7-4d0c-bd9d-0441a8541179

https://github.com/user-attachments/assets/8dbbbe2d-1c9b-4f39-b3ef-84c71b59dff6

### CODE REVIEW

File: Where_Is_My_Wife/Assets/_Main/Scripts/Dialogue/DialogueManager.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
+using TMPro;
+using UnityEngine;
+using WhereIsMyWife.Dialogue;
+
+namespace WhereIsMyWife.Managers
+{
+   public class DialogueManager : Singleton<DialogueManager>
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Dialogue/DialogueManager.cs:7]: I think this system is far to complex to what we may be using, localization will not be necessary for this project, not in scope, remove it, we can figure an easier solution in the near future.

File: Where_Is_My_Wife/Assets/_Main/Scripts/Input/InputSources.cs [hunk 2]
```diff
@@ -15,6 +15,19 @@ public interface IPlayerInputEvent
         Action LookUpAction { get; set; }
         Action<bool> LookDownAction { get; set; }
     }
+
+    public interface ISpecialInputEvent
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Input/InputSources.cs:19]: Why special and not part of the UI?

File: Where_Is_My_Wife/Assets/_Main/Scripts/UI/MultilingualText.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
+using System;
+using TMPro;
+using UnityEngine;
+using WhereIsMyWife.Managers;
+
+public class MultilingualText : MonoBehaviour
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/UI/MultilingualText.cs:6]: Not needed

File: Where_Is_My_Wife/Assets/_Main/Scripts/UI/Settings/LanguageConfig.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+using WhereIsMyWife.Managers;
+
+namespace WhereIsMyWife.Config
+{
+    public class LanguageConfig : ConfigSelection
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/UI/Settings/LanguageConfig.cs:5]: Not needed

File: Where_Is_My_Wife/Assets/_Main/Scripts/UI/Settings/SliderConfig.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+namespace WhereIsMyWife.Config
+{
+    public class SliderConfig : ConfigSelection
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/UI/Settings/SliderConfig.cs:7]: Dont use short words, use full ones, write Configuration or Setting, I prefer Setting

File: Where_Is_My_Wife/Assets/_Main/Scripts/UI/Settings/ToggleConfig.cs [hunk 6]
```diff
@@ -0,0 +1,15 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace WhereIsMyWife.Config  
+{
+    public class ToggleConfig : MonoBehaviour
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/UI/Settings/ToggleConfig.cs:6]: Same

File: Where_Is_My_Wife/Assets/_Main/Scripts/UI/UnityEventText.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
+using TMPro;
+using UnityEngine;
+
+namespace WhereIsMyWife.UI
+{
+    public class UnityEventText : MonoBehaviour
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/UI/UnityEventText.cs:6]: This is not needed, assign text directly

File: Where_Is_My_Wife/Assets/_Utilities/_Ferchus/StringArray.cs [hunk 8]
```diff
@@ -0,0 +1,10 @@
+using System;
+using UnityEngine;
+
+[Serializable]
+public class StringArray
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Utilities/_Ferchus/StringArray.cs:5]: Not needed

**Review by Horothenic — CHANGES_REQUESTED** [review]

File: Where_Is_My_Wife/Assets/_Main/Scripts/Input/InputSources.cs (hunk 2 above)
Comment (FerchusGames) [reply Where_Is_My_Wife/Assets/_Main/Scripts/Input/InputSources.cs]: 7307ce77c324c327dc42980235a42a5c63fb742b

File: Where_Is_My_Wife/Assets/_Main/Scripts/UI/Settings/SliderConfig.cs (hunk 5 above)
Comment (FerchusGames) [reply Where_Is_My_Wife/Assets/_Main/Scripts/UI/Settings/SliderConfig.cs]: ecad7bd72251258c3aa60f443552127fd5eef85c

File: Where_Is_My_Wife/Assets/_Main/Scripts/UI/Settings/ToggleConfig.cs (hunk 6 above)
Comment (FerchusGames) [reply Where_Is_My_Wife/Assets/_Main/Scripts/UI/Settings/ToggleConfig.cs]: ecad7bd72251258c3aa60f443552127fd5eef85c

File: Where_Is_My_Wife/Assets/_Utilities/_Ferchus/StringArray.cs (hunk 8 above)
Comment (FerchusGames) [reply Where_Is_My_Wife/Assets/_Utilities/_Ferchus/StringArray.cs]: 96665c08c08aa3776ec96d7030a7f19c1066d17c

File: Where_Is_My_Wife/Assets/_Main/Scripts/UI/Settings/LanguageConfig.cs (hunk 4 above)
Comment (FerchusGames) [reply Where_Is_My_Wife/Assets/_Main/Scripts/UI/Settings/LanguageConfig.cs]: 96665c08c08aa3776ec96d7030a7f19c1066d17c

File: Where_Is_My_Wife/Assets/_Main/Scripts/UI/MultilingualText.cs (hunk 3 above)
Comment (FerchusGames) [reply Where_Is_My_Wife/Assets/_Main/Scripts/UI/MultilingualText.cs]: 96665c08c08aa3776ec96d7030a7f19c1066d17c

File: Where_Is_My_Wife/Assets/_Main/Scripts/Dialogue/DialogueManager.cs (hunk 1 above)
Comment (FerchusGames) [reply Where_Is_My_Wife/Assets/_Main/Scripts/Dialogue/DialogueManager.cs]: 96665c08c08aa3776ec96d7030a7f19c1066d17c

File: Where_Is_My_Wife/Assets/_Main/Scripts/UI/UnityEventText.cs (hunk 7 above)
Comment (FerchusGames) [reply Where_Is_My_Wife/Assets/_Main/Scripts/UI/UnityEventText.cs]: 090f3bf16c54edf415f841319449c7e9225e904c

**Review by Horothenic — APPROVED** [review]

**Pull request merged**